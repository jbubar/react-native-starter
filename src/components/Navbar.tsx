import React from 'react';

import {
  Divider,
  TopNavigation,
  TopNavigationProps,
} from '@ui-kitten/components';

export default (props: TopNavigationProps) => (
  <>
    <TopNavigation alignment="center" {...props} />
    <Divider />
  </>
);
