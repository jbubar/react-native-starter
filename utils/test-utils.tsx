// test-utils.js

import React, {ReactElement} from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';

import * as eva from '@eva-design/eva';
import {EvaIconsPack} from '@ui-kitten/eva-icons';
import {ApplicationProvider, IconRegistry} from '@ui-kitten/components';

import {render, RenderOptions} from '@testing-library/react-native';

const styles = StyleSheet.create({
  safeArea: {flex: 1},
});

const AllTheProviders: React.FC = ({children}) => {
  return (
    <>
      <IconRegistry icons={EvaIconsPack} />
      <ApplicationProvider {...eva} theme={eva.light}>
        <SafeAreaView style={styles.safeArea}>{children}</SafeAreaView>
      </ApplicationProvider>
    </>
  );
};

const customRender = (ui: ReactElement, options: RenderOptions) =>
  render(ui, {wrapper: AllTheProviders, ...options});

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
