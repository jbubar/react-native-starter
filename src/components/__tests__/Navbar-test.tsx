import React from 'react';
import {render} from 'test-utils';

import Navbar from '../Navbar';

describe('Navbar', () => {
  test('Snapshot', () => {
    const {toJSON} = render(<Navbar title="Sample Title" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
