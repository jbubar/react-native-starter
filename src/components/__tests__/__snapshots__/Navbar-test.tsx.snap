// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "minHeight": 56,
            "paddingHorizontal": 8,
            "paddingVertical": 8,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "zIndex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
              },
              Object {
                "color": "#222B45",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          Sample Title
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "zIndex": 1,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      appearance="default"
      style={
        Array [
          Object {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          undefined,
        ]
      }
    />
  </RCTSafeAreaView>
</View>
`;
